---
title: "Analysis of TRF2 heart Mito Unlabeled data"
author: "Scott Lyons"
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
    code_folding: hide
  pdf_document: default
---

## Background

Data was searched with fragpipe and data analyzed by MSstats.

Deb recently gave me a list of things she would like to see for the output of this analysis. listed below:

-   Filter for mito proteins
-   Filter for TRF/AL q\<.10 in either genotype
-   run the correlation and graph results (TRF/AL WT vs. TRF/AL KO)
-   Generate a list of proteins that are significant (q\<.1) in one genotype but NOT the other, OR changing sig in opposite directions.
-   Highlight those protein in the graph - color coded based on genotype.


### load packages

Load required packages and load data. Process and filter the PSMs data and meta data
```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(MSstats)
library(data.table)   
library(ggprism)
library(ggrepel)
library(openxlsx)
library(pathfindR)
library(pheatmap)
```


### read in data
read in MSstats data and combined protein data.

```{r}
#| include = FALSE

df = fread('data/20221028_TRF2_hrtMito_unlabeled_fragpipe_unique_reps_MSstats.csv')


df = df %>%
  mutate(Condition = if_else(Condition == 'exp', 'Bdh1KO_TRF', Condition))

df$ProteinName = factor(df$ProteinName)
df$PeptideSequence = factor(df$PeptideSequence)

# this is the protein .tsv output file from Fragpipe. I use it to add annotation to the msstats data
protein = fread("data/20221028_TRF2_hrtMito_unlabeled_fragpipe_unique_reps_combined_protein.tsv")


```

### Process data
cleans, normalizes and summarizes the data.

```{r message=FALSE, warning=FALSE, results='hide'}

processed_hrt = dataProcess(df, normalization = "equalizeMedians",  censoredInt = "NA", use_log_file = F)

```

### Perform statistical analysis
create comparison table and perform group comparison

```{r, warning=FALSE, results='hide', eval=FALSE}

levels(processed_hrt$ProteinLevelData$GROUP)

comparison1<-matrix(c(1,0,-1,0),nrow=1)
comparison2<-matrix(c(0,0,-1,1),nrow=1)
comparison3<-matrix(c(-1,1,0, 0),nrow=1)
comparison4<-matrix(c(0,1,0, -1),nrow=1)

compare<-rbind(comparison1, comparison2, comparison3, comparison4)
row.names(compare)<-c("BDH1KO_AL-WT_AL", "WT_TRF-WT_AL", "BDH1KO_TRF-BDH1KO_AL", "BDH1KO_TRF-WT_TRF")
colnames(compare) = c("BDH1KO_AL", "BDH1KO_TRF", "WT_AL", "WT_TRF")


test = groupComparison(compare, processed_hrt, use_log_file = F)

```


```{r, warning=FALSE, results='hide'}

levels(processed_hrt$ProteinLevelData$GROUP)

#comparison1<-matrix(c(1,0,-1,0),nrow=1)
comparison2<-matrix(c(0,0,-1,1),nrow=1)
comparison3<-matrix(c(-1,1,0, 0),nrow=1)
#comparison4<-matrix(c(0,1,0, -1),nrow=1)

compare<-rbind(comparison2, comparison3)
row.names(compare)<-c("WT_TRF-WT_AL", "BDH1KO_TRF-BDH1KO_AL")
colnames(compare) = c("BDH1KO_AL", "BDH1KO_TRF", "WT_AL", "WT_TRF")


test = groupComparison(compare, processed_hrt, use_log_file = F)

```

### Reformat data
reformat the groupComparison output from long to wide to make it easier to read. Also add mitocarta and protein descriptions to the data table.

```{r}
mitocarta = read.delim('db/Mouse.MitoCarta3.0.txt') %>%
  dplyr::select(gene_symbol = Symbol, carta_accession = accession,  MitoCarta3.0_SubMitoLocalization, MitoCarta3.0_MitoPathways) 

add_cols = protein %>%
  dplyr::select(., `Protein ID` , `Gene`, `Description`, `Organism`, `Top Peptide Probability`, contains("Combined")) %>%
  distinct(., `Protein ID`, .keep_all = T)


int = test$ComparisonResult %>%
  dplyr::select(., any_of(c('Protein','Label', 'log2FC', 'pvalue', 'adj.pvalue'))) %>%
  drop_na(., pvalue) %>%
  pivot_wider(names_from = c(Label), values_from = c(log2FC, pvalue, adj.pvalue)) %>%
  separate(Protein, c('db' ,'accession', 'gene_symbol'), sep = "\\|") %>%
  left_join(., add_cols, by = c('accession' = 'Protein ID')) %>%
  left_join(., mitocarta, by = c("Gene" = "gene_symbol")) %>%
  mutate(mitocarta_gene = Gene %in% mitocarta$gene_symbol | grepl('mitochondrial', Description) | accession %in% mitocarta$carta_accession) %>%
  dplyr::select(., gene_symbol = Gene, accession, Organism, Description, mitocarta_gene, contains('MitoCarta'), 
                `Top Peptide Probability`, contains("Combined"), contains('FC') | contains('pvalue'))


#write.xlsx(int, paste(Sys.Date(), 'TRF2_heartMito_FRAGPIPE_unlabeled_noReps.xlsx'), sep = '_')

```


```{r}

mitocarta = read.delim('db/Mouse.MitoCarta3.0.txt') %>%
  dplyr::select(gene_symbol = Symbol, carta_accession = accession,  MitoCarta3.0_SubMitoLocalization, MitoCarta3.0_MitoPathways) 

add_cols = protein %>%
  dplyr::select(., `Protein ID` , `Gene`, `Description`, `Organism`, `Top Peptide Probability`, contains("Combined")) %>%
  distinct(., `Protein ID`, .keep_all = T)


meta = data.frame(file_name = sort(unique(processed_hrt$ProteinLevelData$originalRUN)),
                  group = c(rep("WT_AL", 5), rep("BDH1KO_AL", 5), rep("WT_TRF", 4), rep("BDH1KO_TRF", 5)),
                  group_rep = c(paste(rep("WT_AL", 5), rep(1:5), sep = "_"),
                                 paste(rep("BDH1KO_AL", 5), rep(1:5), sep = "_"),
                                 paste(rep("WT_TRF", 4), rep(1:4), sep = "_"),
                                 paste(rep("BDH1KO_TRF", 5), rep(1:5), sep = "_")))


quant = processed_hrt$ProteinLevelData %>%
  dplyr::select(., c('originalRUN', 'Protein', 'LogIntensities', 'GROUP')) %>%
  left_join(., meta, by = c('originalRUN' = 'file_name')) %>%
  dplyr::select(., -any_of(c('originalRUN', 'GROUP', 'group', 'ID'))) %>%
  pivot_wider(names_from = group_rep, values_from = c(LogIntensities), names_prefix = 'abundance_') %>%
  separate(Protein, c('db' ,'accession', 'gene_symbol'), sep = "\\|") %>%
  left_join(., add_cols, by = c('accession' = 'Protein ID')) %>%
  left_join(., mitocarta, by = c("Gene" = "gene_symbol")) %>%
  #mutate(mitocarta_gene = gene_symbol %in% mitocarta$gene_symbol) %>%
  dplyr::select(., gene_symbol = Gene, accession, Description, contains('mitocarta'), where(is.numeric))



```



# filtering for mito only proteins and filtering for TRF/AL q.value \< 0.1

```{r, eval=FALSE}

reduced = int %>%
  filter(., mitocarta_gene == TRUE)%>%
  filter(., `adj.pvalue_WT_TRF-WT_AL` < 0.1 | `adj.pvalue_BDH1KO_TRF-BDH1KO_AL` < 0.1)

write.xlsx(reduced, paste(Sys.Date(), 'TRF2_heartMito_FRAGPIPE_mitoFiltered_trfAL_0.1.xlsx'), sep = '_')


```


## Scatter plot of TRF/AL WT vs. TRF/AL KO

```{r}

new_lst = int %>%
  mutate(sig_single_genotype = (`adj.pvalue_WT_TRF-WT_AL` < 0.1 & `adj.pvalue_BDH1KO_TRF-BDH1KO_AL` > 0.1) |
            (`adj.pvalue_WT_TRF-WT_AL` > 0.1 & `adj.pvalue_BDH1KO_TRF-BDH1KO_AL` < 0.1)) %>%
  mutate(sig_both_genotype = (`adj.pvalue_WT_TRF-WT_AL` < 0.1 & `adj.pvalue_BDH1KO_TRF-BDH1KO_AL` < 0.1)) %>%
  mutate(opposite_directions = (`log2FC_WT_TRF-WT_AL` > 0 & `log2FC_BDH1KO_TRF-BDH1KO_AL` < 0 |
                                    `log2FC_WT_TRF-WT_AL` < 0 & `log2FC_BDH1KO_TRF-BDH1KO_AL` > 0 )) %>%
  mutate(opposite_directions_and_1gt_sig = sig_single_genotype == T & opposite_directions == T) %>%
  mutate(opposite_directions_and_2gt_sig = sig_both_genotype == T & opposite_directions == T)
  

#write.xlsx(new_lst, paste0(Sys.Date(), '_TRF2_heartMito_FRAGPIPE_different_by_genotype_or_FC_direction.xlsx'))


      ggplot(new_lst, aes(x = `log2FC_WT_TRF-WT_AL`, y = `log2FC_BDH1KO_TRF-BDH1KO_AL`, label = gene_symbol))+
          geom_point(data = new_lst %>% filter(., `adj.pvalue_WT_TRF-WT_AL` < 0.1 & `adj.pvalue_BDH1KO_TRF-BDH1KO_AL` > 0.1),
                     aes(x = `log2FC_WT_TRF-WT_AL`, y = `log2FC_BDH1KO_TRF-BDH1KO_AL`),
                     shape = 21, size = 2.5, color = "black", fill = "#0000C0", stroke = 1)+
          geom_point(data = new_lst %>% filter(., `adj.pvalue_WT_TRF-WT_AL` > 0.1 & `adj.pvalue_BDH1KO_TRF-BDH1KO_AL` < 0.1),
                     aes(x = `log2FC_WT_TRF-WT_AL`, y = `log2FC_BDH1KO_TRF-BDH1KO_AL`),
                     shape = 21, size = 2.5, color = "black", fill = "#A00000", stroke = 1)+
          geom_text_repel(data = new_lst %>% filter(., `adj.pvalue_WT_TRF-WT_AL` < 0.1 & `adj.pvalue_BDH1KO_TRF-BDH1KO_AL` > 0.1),
                     aes(label = gene_symbol))+
          geom_text_repel(data = new_lst %>% filter(., `adj.pvalue_WT_TRF-WT_AL` > 0.1 & `adj.pvalue_BDH1KO_TRF-BDH1KO_AL` < 0.1),
                     aes(label = gene_symbol))+
          theme_prism() +
          xlim(c(-2, 2))+
          ylim(c(-2.5, 2.5))+
          geom_hline(yintercept = 0)+
          geom_vline(xintercept = 0) +
          labs(main = 'Correlation bewteen Genotypes and TRF/AL conditions')

#ggsave(filename = "TRF2_hrtMito_genotype_by_condition_bivariate.pdf", plot = plt2, width = 6, height = 5)

```

### Generate a list of proteins that are significant 
(q.value \< 0.1) in one genotype but NOT the other, OR significantly changing in opposite directions

```{r, eval=FALSE}


new_lst = int %>%
  mutate(sig_single_genotype = (`adj.pvalue_WT_TRF-WT_AL` < 0.1 & `adj.pvalue_BDH1KO_TRF-BDH1KO_AL` > 0.1) |
            (`adj.pvalue_WT_TRF-WT_AL` > 0.1 & `adj.pvalue_BDH1KO_TRF-BDH1KO_AL` < 0.1)) %>%
  mutate(sig_both_genotype = (`adj.pvalue_WT_TRF-WT_AL` < 0.1 & `adj.pvalue_BDH1KO_TRF-BDH1KO_AL` < 0.1)) %>%
  mutate(opposite_directions = (`log2FC_WT_TRF-WT_AL` > 0 & `log2FC_BDH1KO_TRF-BDH1KO_AL` < 0 |
                                    `log2FC_WT_TRF-WT_AL` < 0 & `log2FC_BDH1KO_TRF-BDH1KO_AL` > 0 )) %>%
  mutate(opposite_directions_and_1gt_sig = sig_single_genotype == T & opposite_directions == T) %>%
  mutate(opposite_directions_and_2gt_sig = sig_both_genotype == T & opposite_directions == T)
  

write.xlsx(new_lst, paste(Sys.Date(), 'TRF2_heartMito_FRAGPIPE_different_by_genotype_or_FC_direction.xlsx', sep = '_'))


```



### generate spreadsheet of data for figure 6 data

```{r}
# the minimal data required for the proteomics data

jn = full_join(quant, int)

sh = lapply(rownames(compare), function(x){
  
  red1 = int %>%
    dplyr::select(accession,  contains(x)) %>%
    drop_na(.) 
  
  red2 = inner_join(quant, red1, by = "accession")
  
})

write.xlsx(sh, paste0("SPL_", Sys.Date(), "_TRF2_hrtMito_final_tables.xlsx"))
write.xlsx(jn, paste0("SPL_", Sys.Date(), "_TRF2_hrtMito_final_tables_ALL_DATA.xlsx"))


```


### volcano plots


```{r fig.width=5.5, fig.height=4.5}
# extra ia vector of additional proteins to annotated on the volcano plot

`%notin%` <- Negate(`%in%`)

to_label = c("Tbrg4", "Hibadh", "Mtco2", "Endog", "Mmab", "Acot9", "Aldh1b1", "Mrpl20", "Acot9")

volcano_plot = function(df, type, interactive, filter_mito, plot_title, save){
  
  mix = lapply(rownames(compare), function(y){
    
    volcano = df %>%
      {if(filter_mito == T) filter(., mitocarta_gene == T) else .} %>%
      dplyr::select(., gene_symbol, contains(y)) %>%
      dplyr::select(gene_symbol, FC = contains("FC"), p_value = starts_with("pvalue"), q_value = starts_with("adj.pvalue"))%>%
      mutate(p_value = -log10(p_value), .keep = c("unused")) %>%
      drop_na(gene_symbol) %>%
      drop_na(FC) %>%
      mutate(q_sig = q_value <= 0.05)
    
    neg_chg = volcano %>%
      filter(q_value < 0.05 & FC < 0) %>%
      summarize(counts  = n())
    
    pos_chg = volcano %>%
      filter(q_value < 0.05 & FC > 0) %>%
      summarize(n())
    
    save_name = paste(y, plot_title)
    
    if(type == "protein"){
      q = ggplot(volcano, aes(x = FC, y = p_value, shape = q_sig, label = gene_symbol, q_value = q_value)) +
        geom_point(shape = 21, size = 2.5, color = 'black', stroke = 1,
                   fill = ifelse(volcano$q_value < 0.05 & volcano$FC > 0, "#A00000",
                          ifelse(volcano$q_value < 0.05 & volcano$FC < 0, "#0000C0", "#D4D4D4")))+
        scale_x_continuous(breaks=seq(min(floor(volcano$FC), na.rm=T), max(ceiling(volcano$FC), na.rm=T), 1), 
                           limits = c(min(floor(volcano$FC), na.rm=T), max(ceiling(volcano$FC), na.rm=T)))+
        scale_y_continuous(breaks=seq(0, max(ceiling(volcano$p_value)), 1), limits = c(0, max(ceiling(volcano$p_value))))+
        theme_prism(base_size = 15)+
        geom_text_repel(data = . %>%
                          mutate(label = ifelse(volcano$q_value < 0.05 & volcano$FC > 0 |
                                                volcano$q_value < 0.05 & volcano$FC < 0, gene_symbol, "")),
                          aes(label = label))+
        #geom_text_repel(data = volcano[volcano$gene_symbol %in% to_label, ], aes(label = gene_symbol))+
        geom_hline(aes(yintercept = 1.3), linetype = "dashed", size = 1.5)+
        labs(title = paste(plot_title, "\n", toupper(y), "\n", sep = " "),
             x = paste("Log2 fold change", "\n", toupper(y)), y = "-log10 p-value",
             shape = "q<0.05")
      
      
      if(interactive == TRUE & save == FALSE){
        w = ggplotly(q, tooltip = 'all')
        print(w)
      }
      else if(interactive == TRUE & save == TRUE){
        w = ggplotly(q, tooltip = 'all')
        saveWidget(w, paste(save_name, ".html"), selfcontained = T, libdir = "lib")
      }
      else if(interactive == FALSE & save == FALSE){
        q1 = q + annotate("text", label = neg_chg, x = min(floor(volcano$FC)) + 0.2, y = max(ceiling(volcano$p_value))) +
          annotate("text", label = pos_chg, x = max(ceiling(volcano$FC)) - 0.2 , y = max(ceiling(volcano$p_value)))
        
        print(q1)
      }
      else if(interactive == FALSE & save == TRUE){
        q1 = q + annotate("text", label = neg_chg, x = min(floor(volcano$FC)) + 0.2, y = max(ceiling(volcano$p_value))) +
          annotate("text", label = pos_chg, x = max(ceiling(volcano$FC)) - 0.2 , y = max(ceiling(volcano$p_value)))
        
        ggsave(q1, filename = paste(save_name, ".pdf"), height = 5, width = 6 , units = "in", bg='transparent')
        while (!is.null(dev.list()))  dev.off()
      }
    }
    
  })
  
}

v_plot = volcano_plot(df=int, type = 'protein', interactive = F, filter_mito = T, plot_title = "TRF2 heartMito Proteomics",  save = F)    


```

Clustering

```{r, eval=FALSE}
 
cluster = function(df, plot_type, plot_title, save){
  
  if(plot_type == 'pca'){
    

    df2 = df %>%
      dplyr::select(., contains('abundance'), -contains('pool')) %>%
      drop_na(.) %>%
      t(.) %>%
      as.data.frame(.)

    df4 = df2 %>%
      rownames_to_column('colnames') %>%
      mutate(colnames = str_remove_all(colnames, "abundance_")) %>%
      left_join(., meta, by = c("colnames" = "group_rep"))

    
    pca = df2 %>%
      prcomp(., scale. = T, center = T)
    
    var = summary(pca)$importance
    
    rm = c('[:digit:]$', 'abundance_', '_')
    
    pcadata = data.frame(pca$x[,1:2], ID = df4$ID, subgroup = df4$group)
    
    p1 = ggplot(pcadata, aes(x = PC1, y = PC2, color = subgroup, label = ID))+ 
      geom_point(size = 3) + 
      #geom_text_repel(show.legend = F) +
      theme_prism(base_size = 12, border = TRUE) +
      scale_x_continuous(limits = c(round(mean(pcadata$PC1) - sd(pcadata$PC1)*3, 0), round(mean(pcadata$PC1) + sd(pcadata$PC1)*3, 0)))+
      scale_y_continuous(limits = c(round(mean(pcadata$PC2) - sd(pcadata$PC2)*3, 0), round(mean(pcadata$PC2) + sd(pcadata$PC2)*3, 0)))+
      labs(title = paste("PCA", "\n", plot_title, sep = " "),
            x = paste("PC1", " ", "(", round((var[2,1]*100), 1), "%", ")", sep = ""), 
            y = paste("PC2", " ", "(", round((var[2,2]*100),1), "%", ")", sep = ""))
    
    if(save == T){
      ggsave(paste("PCA", plot_title, ".pdf"), height = 6, width = 12 , units = "in", bg='transparent')
      while (!is.null(dev.list()))  dev.off()
    }
    else{print(p1)}
    
  }
  
  if(plot_type == 'hclust'){
    
    df2 = df %>%
      dplyr::select(., contains('abundance'), -contains('pool')) 
    
    #colnames(df2) = c(paste(meta$group, meta$ID, sep = "_" ))
    colnames(df2) = meta$group_rep

    dist1 = df2 %>%
      drop_na(.) %>%
      t(.) %>%
      dist(.) %>%
      hclust(., method = 'ward.D2') %>%
      as.dendrogram() %>%
      #$dendextend::set("labels_col",  k=4) %>%
      plot(., main = paste("Hierarchical Clustering", '\n', plot_title))
    
  }
}


clus = cluster(df=quant, plot_type = 'pca', plot_title = 'SHR Cohort 2 Proteomics', save = F)

clus = cluster(df=quant, plot_type = 'hclust', plot_title = 'SHR Cohort 2 Proteomics', save = F)


```


### pathway analysis
using pathfindR, and using gene set list from msigdb, collection C2, reactome. 

```{r}

# the author of the package made it clear that the q.value is to be used in the analysis, NOT the p.value

# for mouse gene_sets use "mmu_KEGG" and for pin_name_path use "mmu_STRING"
# also for mouse, custom_genes = mmu_kegg_genes and custom_descriptions = mmu_kegg_descriptions

#mmu_kegg = pathfindR::get_gene_sets_list(source = "KEGG", org_code = 'mmu')

#mmu_react = pathfindR::get_gene_sets_list(source = "Reactome")

mmu_msigdb = pathfindR::get_gene_sets_list(source = "MSigDB", species = "Mus musculus", collection = "C2", subcollection = "CP:REACTOME" )

mmu_genes = mmu_msigdb$gene_sets
mmu_desc = mmu_msigdb$descriptions
#path_to_pin_file = get_pin_file(org = "Rattus_norvegicus")

pathfind = function(df, export_name, export_data = FALSE, save_plot = FALSE){
  
  layer1 = map(rownames(compare), function(y){
    
    int1 = df %>%
      dplyr::select(., gene_symbol, contains(y)) %>%
      dplyr::select(Gene_symbol = gene_symbol, logFC = contains('FC'), 
                    FDR_p = contains('adj.pvalue')) %>%
      drop_na(.) %>%
      as.data.frame()
    
    
    if(nrow(filter(int1, FDR_p < 0.1)) < 5){
      print("not enough proteins with a q.value < 0.1")
      
    }
    else{
      
      output = run_pathfindR(input = int1,
                               convert2alias = FALSE,
                               gene_sets = "Custom",
                               pin_name_path = "STRING",
                               #pin_name_path = path_to_pin_file,
                               custom_genes = mmu_genes,
                               custom_descriptions = mmu_desc,
                               visualize_enriched_terms = F,
                               p_val_threshold = 0.1,
                               plot_enrichment_chart = F,
                               silent_option = F,
                               n_processes = 2)
      
      
      if(export_data == TRUE){

        expt = list("grouped_term_output" = output,
                    "sample_term_values" = int1)
        
        write.xlsx(expt, file = paste0(Sys.Date(),"_pathfindR_results_", export_name, y, ".xlsx"))
        
      }

      return(output)
    }
    
  })
  
}

path_analysis = pathfind(df=int, export_name = "TRF2_hrtMito_unlabeled", export_data = F)

```

### Making the dotplot

```{r fig.width=12, fig.height=5.5}

enricment_chart2 = function(result_df, plot_title, top_terms = 20, plot_by_cluster = FALSE, num_bubbles = 4, even_breaks = TRUE, save_plot){ 
  
  plt1 = map2(result_df, rownames(compare), function(x,y){
    
    necessary <- c("Term_Description", "Fold_Enrichment", 
                   "lowest_p", "Up_regulated", "Down_regulated")
    if (!all(necessary %in% colnames(x))) {
      warning("The input data frame must have the columns:\n", 
           paste(necessary, collapse = ", "))
    }
    if (!is.logical(plot_by_cluster)) {
      warning("`plot_by_cluster` must be either TRUE or FALSE")
    }
    if (!is.numeric(top_terms) & !is.null(top_terms)) {
      warning("`top_terms` must be either numeric or NULL")
    }
    if (!is.null(top_terms)) {
      if (top_terms < 1) {
        warning("`top_terms` must be > 1")
      }
    }
    result_df <- x[order(x$lowest_p), ]
    if (!is.null(top_terms)) {
      if (plot_by_cluster & "Cluster" %in% colnames(result_df)) {
        keep_ids <- tapply(result_df$ID, result_df$Cluster, 
                           function(x) x[seq_len(min(top_terms, length(x)))])
        keep_ids <- unlist(keep_ids)
        result_df <- result_df[result_df$ID %in% keep_ids, 
        ]
      }
      else if (top_terms < nrow(result_df)) {
        result_df <- result_df[seq_len(top_terms), ]
      }
    }
    num_genes <- vapply(result_df$Up_regulated, function(x) length(unlist(strsplit(x, ", "))), 1)
    num_genes <- num_genes + vapply(result_df$Down_regulated, 
                                    function(x) length(unlist(strsplit(x, ", "))), 1)
    
    result_df$Term_Description <- factor(result_df$Term_Description, levels = rev(unique(result_df$Term_Description)))
    log_p <- -log10(result_df$lowest_p)
    
  
    g <- ggplot2::ggplot(result_df, ggplot2::aes_(~Fold_Enrichment, ~reorder(Term_Description, Fold_Enrichment))) +
      geom_point(ggplot2::aes(color = log_p, size = num_genes), na.rm = TRUE) +
      theme_bw()+
      ggplot2::xlab("Fold Enrichment") +
      ggplot2::labs(title = paste(plot_title, "\n", toupper(y), "\n", sep = " ")) +
      ggplot2::scale_color_continuous(low = "dodgerblue1", high = "firebrick1")+ # originally low = "gray75", high = "red1"
      ggplot2::labs(size = "# genes", color = expression(-log[10](p))) +
      ggplot2::theme(axis.title.y = ggplot2::element_blank(), legend.title = element_text(),
                     axis.text.y = ggplot2::element_text(color="black"),
                     axis.text.x = ggplot2::element_text(color="black"))
    
    if (max(num_genes) < num_bubbles) {
      g <- g + ggplot2::scale_size_continuous(breaks = seq(0, max(num_genes)))
    }
    else {
      if (even_breaks) {
        brks <- base::seq(0, max(num_genes), round(max(num_genes)/(num_bubbles + 
                                                                     1)))
      }
      else {
        brks <- base::round(base::seq(0, max(num_genes), 
                                      length.out = num_bubbles + 1))
      }
      g <- g + ggplot2::scale_size_continuous(breaks = brks)
    }
    if (plot_by_cluster & "Cluster" %in% colnames(result_df)) {
      g <- g + ggplot2::facet_grid(result_df$Cluster ~ ., scales = "free_y", 
                                   space = "free", drop = TRUE)
    }
    else if (plot_by_cluster) {
      message("For plotting by cluster, there must a column named `Cluster` in the input data frame!")
    }
    g <- g + ggplot2::theme(panel.grid.major = element_line(size=0.05), panel.grid.minor = element_line(size=0.05))
    
    if(save_plot == T){
      g <- ggsave(paste("bubble_plot", export_name, ".pdf"), units = "in", bg='transparent')
      while (!is.null(dev.list()))  dev.off()
    }
    else{print(g)}
  
  }

)}

plot1 = enricment_chart2(result_df = path_analysis, plot_title = "TRF2_hrtMito_unlabeled", save_plot = F)


```

### MRR heatmap

Deb has sent in an email that she wants to be comparing the FC of TRF/AL changes in WT vs KO. "Should be two column heatmap showing mostly unchanged or upregulated FC vs many downregulated in . Deb sent a spreadsheet of all the proteins she wants pulled that are related to the MRR pathway.


```{r fig.width=3.5, fig.height=5.5}

mrr_proteins = read.xlsx("trf2_hrtMito_data/Copy of 2022-09-22_TRF2_heartMito_FRAGPIPE_different_by_genotype_or_opposite_direction.dm edits (003).xlsx", sheet = 2)

dt2 = int %>%
  filter(accession %in% mrr_proteins$accession) %>%
  dplyr::select(., gene_symbol, "FC" =  "log2FC_WT_TRF-WT_AL", "mKO" = "log2FC_BDH1KO_TRF-BDH1KO_AL") %>%
  column_to_rownames(., "gene_symbol")
  
#pheatmap(dt2, clustering_method = "ward.D2")

#pdf("heatmap_mitoRibo_translation_proteins.pdf", width = 3.5, height = 5.5)
pheatmap(dt2, clustering_method = "ward.D2", color=colorRampPalette(c("#0000C0", "#D4D4D4", "#A00000"))(50))
#dev.off()
```


